core/schema/envfile.go

```diff
@@ -0,0 +1,90 @@
+package schema
+
+import (
+	"context"
+
+	"github.com/dagger/dagger/core"
+	"github.com/dagger/dagger/dagql"
+)
+
+type envfileSchema struct{}
+
+var _ SchemaResolvers = &envfileSchema{}
+
+func (s envfileSchema) Install(srv *dagql.Server) {
+	dagql.Fields[*core.Query]{
+		dagql.Func("envFile", s.newEnvFile).
+			Doc("Initialize an environment file"),
+	}.Install(srv)
+
+	dagql.Fields[*core.EnvFile]{
+		dagql.Func("variables", s.variables).
+			Doc("Return all variables"),
+		dagql.Func("withVariable", s.withVariable).
+			Doc("Add a variable").
+			Args(
+				dagql.Arg("name").
+					Doc("Variable name"),
+				dagql.Arg("value").
+					Doc("Variable value"),
+			),
+		dagql.Func("withoutVariable", s.withoutVariable).
+			Doc("Remove all occurrences of the named variable").
+			Args(
+				dagql.Arg("name").
+					Doc("Variable name to remove"),
+			),
+		dagql.Func("variable", s.variable).
+			Doc("Lookup a variable by name (last occurrence wins)").
+			Args(
+				dagql.Arg("name").
+					Doc("Variable name to lookup"),
+			),
+		dagql.Func("file", s.file).
+			Doc("Return the contents as a file"),
+	}.Install(srv)
+
+	dagql.Fields[*core.File]{
+		dagql.Func("asEnvFile", s.asEnvFile).
+			Doc("Parse as an env file"),
+	}.Install(srv)
+}
+
+func (s envfileSchema) newEnvFile(ctx context.Context, q *core.Query, args struct{}) (*core.EnvFile, error) {
+	return &core.EnvFile{}, nil
+}
+
+func (s envfileSchema) variables(ctx context.Context, obj *core.EnvFile, args struct{}) (dagql.Array[core.EnvVariable], error) {
+	return obj.Variables(), nil
+}
+
+func (s envfileSchema) withVariable(ctx context.Context, obj *core.EnvFile, args struct {
+	Name  dagql.String
+	Value dagql.String
+}) (*core.EnvFile, error) {
+	return obj.WithVariable(args.Name.String(), args.Value.String(), false), nil
```

Comment:
Why expend is set to `false` and isn't a parameter? Is that on purpose?

Reply:
It's a bit tricky though, I'm not sure we want to `expand` the variable when it got registered, but instead only when it's read.

If you convert `EnvFile` to a file, we should keep the original syntax like `BAR=${FOO}` stay that way but if you do `EnvFile.EnvVariable("BAR")` or `EnvFile.EnvVariables` then we should interpolate.

Wdyt @shykes?